tests/Test/Xrefcheck/URIParsingSpec.hs

```diff
@@ -0,0 +1,47 @@
+{- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+module Test.Xrefcheck.URIParsingSpec where
+
+import Universum
+
+import Test.Hspec (Spec, describe, it)
+import Test.HUnit (assertBool)
+import URI.ByteString qualified as URIBS
+
+parse :: Text -> Either URIBS.URIParseError (URIBS.URI)
+parse link = URIBS.parseURI URIBS.laxURIParserOptions (encodeUtf8 link)
+
+spec :: Spec
+spec = do
+  describe "URI parsing should be successful" $ do
+    it "Without the special characters in the query strings" $
+      forM_ withoutSpecChars $ \link -> do
+        assertBool "Failed to parse the link" (isRight $ parse link)
```

Comment:
Also, let's not just check that the parse is successful (with `isRight`).

Let's check that it does indeed return the correct URI. I think you should be able to use the `uri` quasiquoter to express what the result should be: https://hackage.haskell.org/package/modern-uri-0.3.4.4/docs/Text-URI-QQ.html